src/hooks/useBorrowRepayModal/Modal/Repay/index.tsx

```diff
@@ -207,24 +207,27 @@ export const RepayForm: React.FC<RepayFormProps> = ({
 };
 
 export interface RepayProps {
-  asset: Asset;
+  token: Token;
+  vToken: Token;
   includeXvs: boolean;
   onClose: () => void;
 }
 
-const Repay: React.FC<RepayProps> = ({ asset, onClose, includeXvs }) => {
+const Repay: React.FC<RepayProps> = ({ token, vToken, onClose, includeXvs }) => {
   const { t } = useTranslation();
   const { account } = React.useContext(AuthContext);
 
-  const vBepTokenContractAddress = unsafelyGetVToken(asset.token.id).address;
+  const {
+    data: { asset },
+  } = useGetUserAsset({ token });
```

Comment:
In some places we switched from asset to token but in others we still use asset? Do they have different definitions?

Reply:
Right now, an asset represents a specific type we have on the front end and that contains a token, its corresponding vToken and some financial values associated to it (reserves, token price etc..). You can find the type definition in the main type file: `src/types/index.ts`